lib/src/io/vm.dart

```diff
@@ -39,6 +39,13 @@ String readFile(String path) {
   }
 }
 
+String readStdin() {
+  var inputBuffer = new StringBuffer();
+  var line;
+  while ((line = io.stdin.readLineSync()) != null) inputBuffer.writeln(line);
+  return inputBuffer.toString();
```

Comment:
It may be faster to use [`StringAccumulatorSink`](https://www.dartdocs.org/documentation/convert/2.0.1/convert/StringAccumulatorSink-class.html) here and consume the data asynchronously.